### PR TITLE
Body component

### PR DESCRIPTION
**Pull request by njons**

### CODE REVIEW

File: src/components/Body.js [hunk 1]
```diff
@@ -5,9 +5,13 @@ import PropTypes from "prop-types";
 
 export default class Body extends React.Component {
   render() {
-    const markSchemeArray = this.props.markSchemeElements.map((element, i) => (
-      <li key={i}> {element.text} </li>
-    ));
+    const markSchemeArray = this.props.markSchemeElements.map(
+      (element, index) => (
+        <li key={index} onClick={() => this.props.markComplete(index)}>
+          {element.text} <p>{element.completed.toString()}</p>
```
Comment (sima-qian) [inline src/components/Body.js]: Perhaps put a comment making clear the `<p>` tag is temporary and will probably be replaced with a `className` that changes based on the `markComplete` value.

File: src/utils/HistoryCaseRevision.functions.js [hunk 2]
```diff
@@ -1,12 +1,33 @@
 const functions = {
   markComplete: (id, prevState) => {
-    if (!prevState.markSchemeElements[id])
+    console.log("this is id", id);
```
Comment (sima-qian) [inline src/utils/HistoryCaseRevision.functions.js]: Do these console logs still need to be here?

File: __test__/HistoryCaseRevision.test.js [hunk 3]
```diff
@@ -9,12 +9,16 @@ const state = {
...
   markSchemeElements: [
     { text: "string", completed: false },
-    { text: "string2", completed: true }
+    { text: "string2", completed: false }
   ]
 };
 
 describe("Ensure markComplete() successfully updates by ID", () => {
   test("valid id should update state", () => {
+    console.log(
```
Comment (sima-qian) [inline __test__/HistoryCaseRevision.test.js]: Is this console log necessary?

**Comment by martingaston** [comment]
would recommend you test this if you haven't already: 

```
markComplete: (id, prevState) => {
    if (!prevState.markSchemeElements[id]) {
      throw new Error("element out of range");
    }
    const markSchemeElements = [...prevState.markSchemeElements];
    markSchemeElements[id].completed = !markSchemeElements.completed;
    return {
        markSchemeElements,
        markSchemeCompleted: prevState.markSchemeCompleted + markSchemeElements[id].completed ? 1 : -1
    }
  }
```

File: src/components/Body.js (hunk 1 above)
Comment (njons) [reply src/components/Body.js]: done

File: src/utils/HistoryCaseRevision.functions.js (hunk 2 above)
Comment (njons) [reply src/utils/HistoryCaseRevision.functions.js]: done

File: src/utils/HistoryCaseRevision.functions.js [hunk 4]
```diff
@@ -1,12 +1,33 @@
...
-    markSchemeElements[id].completed = true;
+    if (!markSchemeElements[id].completed) {
+      markSchemeElements[id].completed = true;
+      console.log(
+        "this is completed after click",
+        markSchemeElements[id].completed
+      );
+      return {
+        markSchemeElements,
+        markSchemeCompleted: prevState.markSchemeCompleted + 1
```
[the remark this answers is not in the document]
Comment (njons) [reply src/utils/HistoryCaseRevision.functions.js]: done

File: __test__/HistoryCaseRevision.test.js (hunk 3 above)
Comment (njons) [reply __test__/HistoryCaseRevision.test.js]: nope. removed.

**Pull request merged**